### PR TITLE
check ENVIRONMENT variable during run

### PR DESCRIPTION
**Pull request by p-rdx**

### CODE REVIEW

File: core/connection.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
 from mongoengine import connect
 
+from os import getenv
 from core.config import HOST, PORT, DB_NAME
 
+if getenv('ENVIRONMENT') == 'docker':
+	HOST = '0.0.0.0'
```
Comment (my-master) [inline core/connection.py]: Should be 4 spaces instead of a tab

File: core/connection.py [hunk 2]
```diff
@@ -1,5 +1,9 @@
 from mongoengine import connect
 
+from os import getenv
 from core.config import HOST, PORT, DB_NAME
 
+if getenv('ENVIRONMENT') == 'docker':
```
Comment (my-master) [inline core/connection.py]: `if getenv('ENVIRONMENT') != 'docker':` ? otherwise mongo fails to be found

File: core/run.py [hunk 3]
```diff
@@ -91,6 +91,11 @@ def run():
 
     import logging
 
+    if getenv('ENVIRONMENT') == 'docker':
+        for skill in chain(MAX_WORKERS, ANNOTATORS, SKILL_SELECTORS, SKILLS):
```
Comment (my-master) [inline core/run.py]: missing import `from itertools import chain`

Comment (my-master) [inline core/run.py]: Shouldn't have `MAX_WORKERS` here, should have `RESPONSE_SELECTORS` and `POSTPROCESSORS`

**Comment by my-master** [comment]
Initial docker build took about 2 hours. Is it normal? Is there any indicator that shows all containers are running? `docker ps` doesn't suit because it shows that the container is running even if it's not finished yet.

File: core/run.py [hunk 4]
```diff
@@ -91,6 +91,11 @@ def run():
 
     import logging
 
+    if getenv('ENVIRONMENT') == 'docker':
```
Comment (my-master) [inline core/run.py]:
```
    if getenv('ENVIRONMENT') != 'docker':
        for service in chain(ANNOTATORS, SKILL_SELECTORS, SKILLS, RESPONSE_SELECTORS, POSTPROCESSORS):
            if not service.get('external', False):
                service['host'] = 'http://0.0.0.0'
                service['url'] = f"{service['host']}:{service['port']}/{service['endpoint']}"
```

File: core/run.py [hunk 5]
```diff
@@ -101,7 +107,7 @@ def run():
     postprocessor = DefaultPostprocessor()
     skill_caller = RestCaller(max_workers=MAX_WORKERS)
     response_selector = ConfidenceResponseSelector()
-    ss_names, ss_urls = zip(*[(annotator['name'], annotator['url']) for annotator in SKILL_SELECTORS])
+    ss_names, ss_urls = zip(*[(annotator['name'], '{}:{}/{}'.format(annotator['host'], annotator['port'], annotator['endpoint'])) for annotator in SKILL_SELECTORS])
```
Comment (my-master) [inline core/run.py]: This logic should be implemented for all services, not only annotators (see my suggested code in the previous comment)

File: core/connection.py [hunk 6]
```diff
@@ -1,5 +1,9 @@
 from mongoengine import connect
 
+from os import getenv
 from core.config import HOST, PORT, DB_NAME
 
+if getenv('LAUNCHING_ENVIRONMENT') != 'docker':
```
Comment (my-master) [inline core/connection.py]: Still too generic name, easy to interfere with other libraries.

File: core/connection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (p-rdx) [reply core/connection.py]: Fixed

File: core/connection.py (hunk 1 above)
Comment (p-rdx) [reply core/connection.py]: Fixed

File: core/connection.py (hunk 2 above)
Comment (p-rdx) [reply core/connection.py]: Fixed

File: core/run.py (hunk 3 above)
Comment (p-rdx) [reply core/run.py]: fixed

Comment (p-rdx) [reply core/run.py]: fixed

File: core/run.py [hunk 7]
```diff
@@ -87,22 +89,32 @@ def run():
...
+    from core.config import MAX_WORKERS, ANNOTATORS, SKILL_SELECTORS, SKILLS, RESPONSE_SELECTORS, \
+        POSTPROCESSORS
 
     import logging
 
+    for service in chain(ANNOTATORS, SKILL_SELECTORS, SKILLS, RESPONSE_SELECTORS, POSTPROCESSORS):
+        if getenv('DPA_LAUNCHING_ENV') != 'docker':
+            if not service.get('external', False):
+                service['host'] = 'http://0.0.0.0'
+        service['url'] = service[
```
Comment (NickVolynkin) [inline core/run.py]: `service['url']  = service['url']`?

Comment (my-master) [reply core/run.py]: will fix, thanks

**Pull request merged**